DEVELOPING.md

```diff
@@ -89,48 +101,71 @@ The repository is organized as follows:
 
 * `ir/` The Intermediate Representation, produced by the compiler and consumed by the linker
 * `compiler/` The scalac compiler plugin
-* `tools/` The linker, optimizer, verifier, etc.: everything that happens at link time
+* `io/` Virtual I/O abstractions
+* `logging/` A tiny logging API
+* `linker/` The linker, optimizer, verifier, etc.: everything that happens at link time
 
 ### Library
 
 * `library/` The Scala.js standard library (everything inside `scala.scalajs.*`)
 * `javalanglib/` Implementation in Scala.js of the classes in `java.lang.*`
 * `javalib/` Implementation in Scala.js of other classes in `java.*`
-* `javalib-ex/` Some more Java classes with non-standard dependencies
 * `scalalib/` Almost void project for recompiling the Scala library for Scala.js
 * `library-aux/` A few files of the Scala library that need to be compiled separately
 
-All of these, except `javalib-ex`, are packaged in `scalajs-library.jar` as part
-of `library/package`.
+All of these are packaged in `scalajs-library.jar`.
 
-### sbt plugin
+### JS environments
 
-Note that the sbt plugin depends on the IR and the tools.
+The JS environments are JVM libraries that abstract the details of using a
+JavaScript engine to run JS code.
 
 * `js-envs/` The generic definitions of JavaScript environments and runners
 * `nodejs-env/` The Node.js environment
+
+Other JS environments are developed in separate repositories under the
+`scala-js` organization.
+
+### Testing infrastructure
+
+There is a generic infrastructure that maps the sbt-testing-interface API
+across the JVM/JS boundary, so that Scala.js testing frameworks can be piloted
+from JVM processes such as sbt.
+
+* `test-interface/` JS side of the bridge
```

Comment:
This also contains the JS specific version of `sbt.testing._`.

Reply:
Indeed. Updated.